File: go/0111E-二叉树的最小深度.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func minDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	left := minDepth(root.Left)
	right := minDepth(root.Right)
	if left == 0 || right == 0 {
		return left + right + 1 // 有一个为 0 ， 相当于存在的子树深度
	} else if left > right {
		return right + 1
	} else {
		return left + 1
	}
}

File: cpp/0111E-二叉树的最小深度.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-21 10:10:45
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    int minDepth(TreeNode* root) {
        if (root == nullptr) return 0;
        int leftDepth = minDepth(root->left);
        int rightDepth = minDepth(root->right);
        if (leftDepth == 0 || rightDepth == 0)
            return leftDepth + rightDepth + 1;
        if (leftDepth < rightDepth)
            return leftDepth + 1;
        else
            return rightDepth + 1;
    }
};